### DISCUSSION TITLE
Can i stop propagation on drop with nested targets?

### DISCUSSION THREAD

**Discussion by irinakk** [atlassian/pragmatic-drag-and-drop#110] [Q&A]
I'm using pdnd for a [nested drop target](https://atlassian.design/components/pragmatic-drag-and-drop/core-package/drop-targets/#nested-drop-targets) scenario, i noticed with [A(canDrop => false), B], the onDrop on B will still be triggered. 

Is there any way i can stop this propagation when A.canDrop fails the prediction?

**Comment by alexreardon** [comment]
From the [drop target documentation](https://atlassian.design/components/pragmatic-drag-and-drop/core-package/drop-targets/):

> Returning `false` from `canDrop()` will not block dropping on parent or child drop targets. All drop
targets that want to not allow dropping, need to return `false` from their `canDrop()` function.
>
> When looking for valid drop targets, a lookup starts from the deepest part of the DOM tree the user
is currently over and searches upwards for valid targets. If a drop target blocks dragging
(`canDrop()` returns `false`), then that drop target is ignored and the search upwards continues.

This approach has been taken for maximum flexibility - each drop target being dragged over has the opportunity to be dropped onto (if it wants).

In a similar setup to what you are describing, there might be experiences were a child cannot be dropped on (due to business logic) but the parent drop target could still accept the drop. One example could be: when dragging a "task" into a "list" that contains tasks - it might be possible to only drop onto a few tasks in the list, but it could always be possible to drag into the list itself. 

Having each drop target be able to make it's own decision about whether it can be dropped on gives consumers full control on what the logic should be for their experience.

So if you want child drop target to block dropping on a parent in a hierarchy to block dropping, they all need to `return false` from `canDrop()`. There are many possible ways you can do this. One example, is that if you have some state managing what can be dropped on, then you can use that state in both places to calculate what to return from `canDrop()`.
